### DISCUSSION TITLE
Re-adjusted Retro ratio

### DISCUSSION THREAD

**Discussion by kdecks** [RetroAchievements/RAWeb#664] [Ideas]
It needs to take in account revisions and additions to the set. 

Also possibly not have the point value tied to the dev value, this would been to be experimented on.

**Comment by julkasalsa** [comment]
Total plays counted per achievement not per game.

**Comment by luchaos** [comment]
See https://github.com/RetroAchievements/RAWeb/pull/504

**Comment by meleu** [comment]
I'm from an opinion that retro-ratio is a feature that should be eliminated.

Listing the pros and cons:

Pros:
- **tries** to be a way to measure how difficult an achievement is
- encourages engagement of competitive players

Cons:
- fails to meet its original intention (as it's not accurate)
- visual polution
- brings confusion to the users
- causes tension in the community when rescore and/or revisions happen (wich are both things that clearly are beneficial for the overall quality of the project)
- can waste skilled people's time when they work on this buggy and dispensable feature.


IMO the cost of having it is not worth it.

---

@Xymjak, as you spent a lot of time thinking and coding that PR, it seems you like such feature.

Could you share your opinions about what I said above? What are the pros and cons of the retro-ratio feature in your opinion?

**Reply by MagmarFire** [reply]
I don't have an issue with Retro Ratio being a thing, personally. My primary opinion is that it should be fixed, however.

But overall, I'm just ambivalent about it at best. If everyone else thinks it should be removed, it won't bug me too much.

**Reply by televandalist** [reply]
I'm not crazy about the idea of axing it when it could be improved. It's an experimental feature. Let's experiment 🤷‍♂️ Maybe start by merging #504 for its one-year anniversary? 

`tries to be a way to measure how difficult an achievement is` - I always think of it more as showing rarity rather than difficulty. Difficulty should be reserved for the actual points, with rarity scaling it as Retro Points.

`encourages engagement of competitive players` - This trumps all the cons, doesn't it?

`fails to meet its original intention (as it's not accurate)` - There's plenty of ways to make it more accurate. See #504 - The biggest thing hurting its accuracy is the fact that achievements that are part of a revision affect the overall set. Having it based on the number of players/unlocks since an achievement's upload date would help a lot. Also, making it so untracked users and softcore don't affect them. 

`visual pollution` - So are hundreds of Mario hacks. Maybe have an option to toggle the visibility? 

`causes tension in the community when rescore and/or revisions happen (which are both things that clearly are beneficial for the overall quality of the project)` - People are going to complain about revisions no matter what. I don't see many people complaining about Retro Points when it comes to revisions anyway. There's plenty of jokes about it, though. 

`can waste skilled people's time when they work on this buggy and dispensable feature.` - The same thing could be said about most things pertaining to V1. Thankfully we're all volunteers who choose how we waste our time 😌

**Reply by meleu** [reply]
> Thankfully we're all volunteers who choose how we waste our time

Indeed. Let's wait and see who of those able to review, merge and maintain that code ([labeled as "spaghetti" by its own author](https://github.com/RetroAchievements/RAWeb/pull/504/files#diff-b9eaf544a2031b36c75df98a0471348a40fdb35ab82da360137e54c7c238d03dR126)) will choose to do so.

But if I could somehow influence them, I'd like to suggest them to invest their time in some other more clearly beneficial issue. Examples: #642, #618, #619, #459, #521 (or anything that could make QA Team's life easier).

**Reply by Xymjak** [reply]
> Could you share your opinions about what I said above? 

Sorry if I'll sound too harsh or angry, I'm bad with English and have no idea how to write correctly~
So, opinions~

---

>  fails to meet its original intention

This argument is wrong and can only be made by people who: 
- don't play a lot
- play only casual sets
- don't pay attention to points at all

RP (not broken) is much more objective that normal points. If I see a 50 points achievement I never can say if it'll take me 30 minutes or 20 hours. But looking at RP (not broken) I can predict the effort time pretty close.

The same for whole sets. RP (not broken) shows the objective difference between a 300 (400) set and a 300 (800) set. Which is also very good for events like Leapfrog or Peaker. Without RP these events will completely turn into "just copypaste what others are doing" because you can't otherwise easily tell yourself if a set will allow you to survive in an event or eliminate you. But if you are "the 1st" you already can't copypaste others.

"But we should rescore everything to be objective" - the current system simply not allows it, we are limited by "5 points for minor significance" below and 50 points above (with rare 100). While the effort between 5 and 50 can be anywhere from 5x to 500x. And expanding normal points range will make casual players (and devs) even more angry, they already want to reduce that range and make easier achievements worth more, they simply can't see a difference between "slightly hard, pretty hard, very hard, brutal hard and impossibly hard". RP (not broken) shows that difference very good.

---

> visual polution
> brings confusion to the users

These 2 "problems" are easily solvable with a off-by-default-switch with explicit explanation tooltip. (I hope in the next ~2 weeks I'll be able to make public the PR which is focused on another problem but as side effect also demonstrates how these 2 "problems" can be solved with a few lines of code.)

---

> causes tension in the community when rescore and/or revisions happen

I'm sorry?.. Rescores and Revisions cause tension by themselves. I can't even remember if somebody ever mention negatively "Retropoints" near "Rescore". And the only negative mentions about "Retropoints" near "Revisions" is "why retropoints are still not fixed?" (IE negative not about them but about ignoring them), and it's still happens much less than "Ban all Revisions"-complain without any mentions of RPs.

IE that argument is wrong...

---

a little off:
> labeled as "spaghetti" by its own author

Because the site is spaghetti...
80% of that code is simple
SELECT ID, NumPlayers FROM `Subsets` WHERE GameId = $gameId
Or something like that (I can think about many different implementation).
That code wouldn't exist if I wasn't told to not change anything db-related in V1.

---

> seems you like such feature

So how I (and not only me) see RP

Pros:
- more objectively shows how much effort required by achievements/sets
- encourages engagement of competitive players
- helps with events and makes them more fun
- helps with planning "what to play and when"

Cons:
- wasting time on trying to explain why RP (not broken) is useful for active players to people who don't want to be active players...
(But for them should be an off-by-default-switch)

---

Again, sorry if I sound too angry~

PS 
> Examples

Clicked the last, coded the last... Will check the rest after another PR~

**Reply by ScottFromDerby** [reply]
hey I didn't write that it's spaghetti (although it is)

**Reply by meleu** [reply]
@Xymjak

> sorry if I sound too angry~

Don't worry. I guess the catalyst for such tone was my mention to the :spaghetti: thing...

Unfortunately I didn't choose the best words, my sincere apologies for that. I didn't mean to belittle the quality of your work. In fact, really appreciate your contributions. The fact that I've upset someone who actively contributes to the project (while the vast majority are just demanding) makes me feel kinda embarrassed.

But, as you said (and Scott confirmed), the whole site is :spaghetti:. And that's why I think we shouldn't use it as an inspiration. Instead, we should strive to decrease the amount of it.

Anyways, abstracting the emotional part. Your input was very useful and made me realize something that I, indeed, wasn't seeing:

RP (not broken) can help participants of events like Leapfrog to smartly choose their games (and also make the events more fun).

This makes me see it with a different point of view.

Although I still think it's not accurate, it can be a useful hint about the amount of effort required to unlock an achievement.

---

Regarding the other point I brought, saying it "brings confusion to the users", I don't think that (optionally) hiding the feature would solve it.

What we need a clear documentation explaining how it works.

Would you be willing to reinforce that con and ~~waste~~ spend some more time explaining why RP (not broken) is cool and how it works?

It would be useful not only for people who don't want to be active players but for the whole community (current and future members).

If English is an issue, I'm sure @pinguupinguu would be glad to help and review the doc. You can use [this page](https://github.com/RetroAchievements/RANews/wiki/How-RetroPoints-works) to start drafting it.

I would volunteer to more carefully review and then merge #504 if we had such doc.

And after merging it we would publish an article in RANews advertising the change.

**Comment by Jamiras** [comment]
Retro Ratio (or any other sort of algorithmically assigned point system) will never work perfectly because of what I've come to call the Chocolate Donut paradox. [Chocolate Donut](http://retroachievements.org/achievement/2305) is a dirt-simple achievement (complete the level normally), but only earned by 17% of players because it's late game and in a very popular game where most players move on before they get that far in the game.

It used to be 5 points with a RR of something like 40. Now it's 1 point to reflect it's difficulty, and the RR is 2.

For this reason alone, I agree that RR should be eliminated.

But there is some merit in identifying which achievements are the most difficult that a player has earned (as reflected by the My Pages > Achievements default sort), and some of the changes in #504 improve the RR calculation to better reflect that.
